### ISSUE TITLE
fillRect with BL_FORMAT_A8 becomes slanted

### ISSUE THREAD

**Issue by williamhCode** [blend2d/blend2d#216]
If I use non-integer values with fillRect and BL_FORMAT_A8, the rect becomes slanted.
```c++
#include <blend2d.h>

int main(int argc, char *argv[]) {
  BLImage img(480, 480, BL_FORMAT_A8);
  BLContext ctx(img);
  ctx.clearAll();
  ctx.fillRect(0, 0, 100.5, 100.5);
  ctx.end();

  img.writeToFile("test.png");

  return 0;
}
```
![test](https://github.com/user-attachments/assets/08cdf926-deaf-479a-b3ba-9dd55ee77d0e)

**Comment by kobalicek** [comment]
Are you using the latest version? I cannot repro this issue, also this can be verified on fiddle.blend2d.com with this snippet:

```c++
#include <blend2d.h>
 
BLImage render(const BLContextCreateInfo& cci) {
  BLImage img(480, 480, BL_FORMAT_A8);
  BLContext ctx(img);
  ctx.clearAll();
  ctx.fillRect(0, 0, 100.5, 100.5);
  ctx.end();
  return img;
}
```

**Comment by williamhCode** [comment]
Yes, I am on the latest main branch with macOS 14.4 (arm64). I tried with both LLVM clang (19.1) and apple clang (15.0) and it seems to produce the same output. I tried it with different dimensions and it gives the same slanted output. If I build with `-D BLEND2D_NO_JIT=ON`, the output is correct, so this has something to do with the JIT compiler I assume.

**Comment by kobalicek** [comment]
Managed to repro it, will look at it, thanks!

**Comment by kobalicek** [comment]
Ok, I looked into it today as I was travelling a lot, and I found the problem, but I have found some other issues related to A8 pixel format, which I'm going to fix as well.

**Comment by kobalicek** [comment]
This issue should be fixed by f6fdedd5f449e5f2895409ac52d28407a2f40766

Thanks a lot for reporting!

BTW there are more issues regarding A8 pixel format, which I'm going to fix next.

**Comment by williamhCode** [comment]
No problem and thanks for the quick fix!
